src/forms/repositorySpec.js

```diff
@@ -1,64 +1,130 @@
+    })
 
-        let formsRepository = new FormsRepository(db);
-
-        let actualError = null;
-        try {
-            await formsRepository.find("9bdf21e7-52b8-4529-991b-5f2df9de0323");
-        } catch(error) {
-            actualError = error;
-        }
+    describe("createForm function", function() {
+        let hutId, formConfigs, expectedFormsArguments;
+        let expectedFormsQuery;
+        beforeEach(function() {
+            hutId = "9bdf21e7-52b8-4529-991b-5f2df9de0323";
+            formConfigs = {
+                showOrgType: false,
+                showBankDetails: false,
+                showEan: false,
+                showCleaningToggle: false,
+                defaultCleaningInclude: false,
+                showArrivalTime: false,
+                showDepartureTime: false,
+                stdArrivalTime: false,
+                stdDepartureTime: false,
+                stdInformation: ""
+            }
+            expectedFormsQuery = `INSERT INTO forms( id, hutId, data) VALUES( $1::uuid, $2::uuid, $3::json)`;
+            expectedFormsArguments = function(id) {             
+                return [
+                    id,
+                    hutId = jasmine.any(String),
+                    {
+                        createdAt: jasmine.any(String),
+                        updatedAt: jasmine.any(String),
+                        ...formConfigs
+                    }
+                ]
+            }
+        })
 
-        expect(actualError).not.toBe(null);
-    });
+        it("creates a form and returns the form's id", async function() {
+            let formsRepository = new FormsRepository(db);
+            let id;
 
-    it("succeeds if a form exists with the provided id", async function() {
-        let foundForm =  {
-            rows: [
-                {
-                    id: "some id",
-                    hutId: "9bdf21e7-52b8-4529-991b-5f2df9de0323",
-                    data: {
-                        showOrgType: false,
-                        showBankDetails: false,
-                        showEan: false,
-                        showCleaningToggle: false,
-                        defaultCleaningInclude: false,
-                        showArrivalTime: false,
-                        showDepartureTime: false,
-                        stdArrivalTime: false,
-                        stdDepartureTime: false,
-                        stdInformation: ""
-                    }
-                }
-            ]
-        };
+            let actualError = null;
+            try {
+                id = await formsRepository.create(hutId, formConfigs);
+            } catch(error) {
+                actualError = error; 
+            }
 
-        let db = {
-            query: jasmine.createSpy("db.query").and.callFake(async () => foundForm)
-        };
+            expect(actualError).toBe(null);
+            expect(db.query).toHaveBeenCalledWith(expectedFormsQuery,expectedFormsArguments(id));
```

Comment:
You never assert that the `id` returned here is correct.

Reply:
hmm i'm not sure how to do that? What is a 'correct' id?
I do however assert that the returned id is also the one that was given as argumet to the database (when i call 'expectedFormsArguments(id)')